pkg/deploy/lattice/target_group_synthesizer.go

```diff
@@ -355,7 +330,59 @@ func (t *targetGroupSynthesizer) isTargetGroupUsedByaHTTPRoute(ctx context.Conte
 	return false
 }
 
+func (t *targetGroupSynthesizer) SynthesizeTriggeredTargetGroupsDeletion(ctx context.Context) error {
+	var resTargetGroups []*latticemodel.TargetGroup
+	var returnErr = false
+
+	t.stack.ListResources(&resTargetGroups)
+
+	glog.V(2).Infof("SynthesizeTriggeredTargetGroupsDeletion: TargetGroups ==[%v]\n", resTargetGroups)
+	for _, resTargetGroup := range resTargetGroups {
+		if !resTargetGroup.Spec.IsDeleted {
+			glog.V(6).Infof("SynthesizeTriggeredTargetGroupsDeletion should ignore target group creation request for tg: [%v]\n", resTargetGroup)
+			continue
+		}
+
+		if resTargetGroup.Spec.Config.IsServiceImport {
+			// For delete TargetGroup request triggered by service import, we just delete the tg in the datastore
+			t.latticeDataStore.DelTargetGroup(resTargetGroup.Spec.Name, resTargetGroup.Spec.Config.K8SHTTPRouteName, resTargetGroup.Spec.Config.IsServiceImport)
+		} else {
+			// For delete TargetGroup request triggered by k8s service, invoke vpc lattice api to delete it, if success, delete the tg in the datastore as well
+			err := t.targetGroupManager.Delete(ctx, resTargetGroup)
+			if err != nil {
+				glog.V(6).Infof("Delete Target Group in SynthesizeTriggeredTargetGroupsDeletion: failed to delete target group %v, err %v \n", resTargetGroup, err)
+				returnErr = true
+			} else {
+				glog.V(6).Infof("Delete Target Group in SynthesizeTriggeredTargetGroupsDeletion: successfully deleted target group %v\n", resTargetGroup)
+				t.latticeDataStore.DelTargetGroup(resTargetGroup.Spec.Name, resTargetGroup.Spec.Config.K8SHTTPRouteName, resTargetGroup.Spec.Config.IsServiceImport)
+			}
+		}
+	}
+	if returnErr {
+		return errors.New(LATTICE_RETRY)
+	} else {
+		return nil
+	}
+}
+
 func (t *targetGroupSynthesizer) PostSynthesize(ctx context.Context) error {
```

Comment:
Are we still using this?  I thought HTTPRoute's synthesizer already takes care of. both K8S HTTPRoute creation and deletion.     

If this is the case, please revert this change.

Reply:
latticeServiceStackDeployer.Deploy() (used by httproute controller) uses the 
SynthesizeTriggeredTargetGroupsDeletion and SynthesizeSDKTargetGroups explcitly

But, latticeTargetGroupStackDeployer.Deploy() (used by  serviceExport controller)  still uses 
targetGroupSynthesizer.PostSynthesize()

Reply:
There is `no` need to change the `targetgroupSynthesize.PostSynthesize()` internal business logic for serviceexport. and also there is `no` need to change `targetgroupSynthesizer.Synthetize() internal business logic too for serviceexport.